apps/prairielearn/elements/pl-drawing/elements.py

```diff
@@ -1969,35 +1969,28 @@ def generate(el, data):
                 y3 = line[2]["y"]
                 curved_line = True
             else:
-                raise Exception(
-                    "pl-graph-line error: the attribute end-points expects a list of size 2 or 3."
-                )
+                msg = "pl-graph-line error: the attribute end-points expects a list of size 2 or 3."
+                raise ValueError(msg)
         else:
-            raise Exception(
-                "pl-graph-line error: required attribute end-points is missing."
-            )
+            msg = "pl-graph-line error: required attribute end-points is missing."
+            raise ValueError(msg)
 
         if "end-gradients" in el.attrib:
             if curved_line:
-                raise Exception(
-                    "pl-graph-line error: The end-gradients attribute conflicts with an end-points attribute of length 3. You should either provide three points to make a curve or the gradient, but not both."
-                )
-            else:
-                grads = json.loads(pl.get_string_attrib(el, "end-gradients"))
-                if len(grads) != 2:
-                    raise Exception(
-                        "pl-graph-line error: the attribute end-gradients expects an array with 2 values, one for each end point."
-                    )
-                grad1 = grads[0]
-                grad2 = grads[1]
-                if abs(grad1 - grad2) < 1e-9:
-                    raise Exception(
-                        "The provided gradients are not compatible to compute a quadratic curve between the given points."
-                    )
-                else:
-                    x3 = ((y2 - grad2 * x2) - (y1 - grad1 * x1)) / (grad1 - grad2)
-                    y3 = (y1 - grad1 * x1) + grad1 * x3
-                    curved_line = True
+                msg = "pl-graph-line error: The end-gradients attribute conflicts with an end-points attribute of length 3. You should either provide three points to make a curve or the gradient, but not both."
+                raise ValueError(msg)
+            grads = json.loads(pl.get_string_attrib(self, "end-gradients"))
```

Comment:
it looks like this leaked over from another rule (likely a PYLINT rule). This was automatically applied, and just removes the else clause since the other branch returns an exception.